### ISSUE TITLE
Worse performance than pyerfa on Windows

### ISSUE THREAD

**Issue by dcherrie-dstl** [JuliaAstro/ERFA.jl#57]
On windows I have found ~2-3X worse performance compared to pyerfa. Using cmakelists (https://github.com/liberfa/erfa/pull/75) and a clang compiler I was able to substitute the dll resulting in comparative (slightly better) performance.

**Comment by helgee** [comment]
Which compiler flags did you use?

**Comment by giordano** [comment]
Looking to the logs of [`ERFA_jll`](https://github.com/JuliaBinaryWrappers/ERFA_jll.jl) (they are in all tarballs), the library was compiled with `-g -O2`.  CMake configured with `-DCMAKE_BUILD_TYPE=Release` instead would typically use `-O3`, that can make quite some difference

**Comment by dcherrie-dstl** [comment]
I used Visual Studio 2019 and set the target to 'x64-Clang-Release'. PSB a snippet of the ninja build output (build.ninja). Not sure if its helpful; I dont have much experience with build systems and compilation.


build src\CMakeFiles\erfa.dir\apcg13.c.obj: C_COMPILER__erfa_RelWithDebInfo ..\..\..\src\apcg13.c || cmake_object_order_depends_target_erfa
  DEFINES = -DHAVE_CONFIG_H -Derfa_EXPORTS
  **FLAGS = -m64 -fdiagnostics-absolute-paths  /DWIN32 /D_WINDOWS /W3 /MD /Zi /O2 /Ob1 /DNDEBUG**
  INCLUDES = -I. -I..\..\..\src
  OBJECT_DIR = src\CMakeFiles\erfa.dir
  OBJECT_FILE_DIR = src\CMakeFiles\erfa.dir
  TARGET_COMPILE_PDB = src\CMakeFiles\erfa.dir\
  TARGET_PDB = src\erfa.pdb

**Comment by helgee** [comment]
Thanks for posting the build lof @dcherrie-dstl. I can see that your binaries were compiled with `-O2` as well. It seems that Clang produces more performant binaries than our GCC cross-compiler. Could this be the case @giordano? I guess there is not much we can do here?

Out of curiosity, if you use the DLL from Clang and run the tests (`import Pkg; Pkg.test("ERFA")`), do they pass?

**Comment by giordano** [comment]
Yes, it's possible that Clang is doing a better job at building more performant binaries, but the factor 2-3x looks a bit surprising.  Maybe GCC is being very conservative about what to optimise?  I think that trying to build with `-O3`, for example by switching to the CMake build system, is worth a try

**Comment by helgee** [comment]
Unfortunately, the CMake build system has not been merged, yet. I will give it a try locally on macOS.

**Comment by giordano** [comment]
Ugh, right, I missed that :sweat_smile:

**Comment by helgee** [comment]
Running the tests on macOS is virtually identical with `-O2` and `-O3`.
